### PR TITLE
First version - roman_r_m

### PR DESCRIPTION
**Pull request by roman-r-m**
#### Check List:
- [x] Tests pass (`./test.sh <username>`)
- [x] All formatting changes by the build are committed
- [x] Output matches that of calculate_average_baseline.sh
* Execution time: 45s
* Execution time of reference implementation: 1m 56s

### CODE REVIEW

**Comment by gunnarmorling** [comment]
There's a difference in test output:

```
Validating calculate_average_roman_r_m.sh -- src/test/resources/samples/measurements-10.txt

Using java version 21.0.1-graal in this shell.

real	0m0.174s
user	0m0.248s
sys	0m0.083s
1c1
< {Adelaide=10.0/15.0/15.0, Cabo San Lucas=10.0/14.9/14.9, Dodoma=10.0/22.2/22.2, Halifax=10.0/12.9/12.9, Karachi=10.0/15.4/15.4, Pittsburgh=9.7/9.7/9.7, Ségou=10.0/25.7/25.7, Tauranga=10.0/38.2/38.2, Xi'an=10.0/24.2/24.2, Zagreb=10.0/12.2/12.2}
---
> {Adelaide=15.0/15.0/15.0, Cabo San Lucas=14.9/14.9/14.9, Dodoma=22.2/22.2/22.2, Halifax=12.9/12.9/12.9, Karachi=15.4/15.4/15.4, Pittsburgh=9.7/9.7/9.7, Ségou=25.7/25.7/25.7, Tauranga=38.2/38.2/38.2, Xi'an=24.2/24.2/24.2, Zagreb=12.2/12.2/12.2}

```

**Comment by roman-r-m** [comment]
Sorry, should be fixed now.

There was some issue with line endings on my machine which caused diff to produce spurious output which in turn masked the real issue.

**Comment by roman-r-m** [comment]
Made some changes, new version takes ~31 seconds on my machine.

**Comment by roman-r-m** [comment]
With the latest changes the time is now down to ~23s

File: calculate_average_roman-r-m.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+JAVA_OPTS="--enable-preview -XX:+UseTransparentHugePages"
+time java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_roman_r_m
```
Comment (gunnarmorling) [inline calculate_average_roman-r-m.sh:19]:
```suggestion
java $JAVA_OPTS --class-path target/average-1.0.0-SNAPSHOT.jar dev.morling.onebrc.CalculateAverage_roman_r_m
```

Not needed any more.

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
There's a test failure. Please run `./test.sh roman-r-m` and make sure there are no diffs:

```
Validating calculate_average_roman-r-m.sh -- src/test/resources/samples/measurements-dot.txt
Exception in thread "main" java.lang.IndexOutOfBoundsException: Out of bound access on segment MemorySegment{ heapBase: Optional.empty address:140236053483520 limit: 18 }; new offset = 12; new length = 8
	at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.outOfBoundException(AbstractMemorySegmentImpl.java:426)
	at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.apply(AbstractMemorySegmentImpl.java:407)
	at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.apply(AbstractMemorySegmentImpl.java:69)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:124)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:448)
	at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.checkBounds(AbstractMemorySegmentImpl.java:396)
	at java.base/jdk.internal.foreign.AbstractMemorySegmentImpl.checkAccess(AbstractMemorySegmentImpl.java:356)
	at java.base/java.lang.invoke.VarHandleSegmentAsLongs.checkAddress(VarHandleSegmentAsLongs.java:81)
	at java.base/java.lang.invoke.VarHandleSegmentAsLongs.get(VarHandleSegmentAsLongs.java:108)
	at java.base/java.lang.foreign.MemorySegment.get(MemorySegment.java:1595)
	at dev.morling.onebrc.CalculateAverage_roman_r_m.main(CalculateAverage_roman_r_m.java:67)
```

File: calculate_average_roman-r-m.sh (hunk 1 above)
Comment (roman-r-m) [reply calculate_average_roman-r-m.sh]: fixed

**Comment by roman-r-m** [comment]
The test should be fixed now

**Comment by roman-r-m** [comment]
Parallelised version now takes ~6.3 seconds

**Comment by gunnarmorling** [comment]
00:10.949. Thx for participating!

**Pull request merged**